CoreFoundation/String.subproj/CFString.c

```diff
@@ -4669,7 +4667,7 @@ CFDataRef CFStringCreateExternalRepresentation(CFAllocatorRef alloc, CFStringRef
 	// if result == length, we always succeed
 	//   otherwise, if result == 0, we fail
 	//   otherwise, if there was a lossByte but still result != length, we fail
-        if ((result != length) && (!result || !lossByte)) return NULL;
+        if ((result != length) && !(result && lossByte)) return NULL;
```

Comment:
This logic is harder to follow. The comments above explain the thought process and the existing code is reasonably readable, while `result && lossByte` is not.

Reply:
This has not been resolved; please revert to the original.